Provisioner TC scripts for data only deployment.

ashukakkar opened the pull request:
Signed-off-by: ashukakkar <karan.kakkar@seagate.com>

# Problem Statement
- Data only pod deployment.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [x] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

ashukakkar commented:
Hi,

Test run HTML report and console logs are attached in below zip file.
[Data only deploy.zip](https://github.com/Seagate/cortx-test/files/8551752/Data.only.deploy.zip)

pragamj commented on comptests/prov/test_prov_data_only_deploy.py:
```diff
@@ -0,0 +1,150 @@
+    @pytest.mark.comp_prov
+    @pytest.mark.tags("TEST-39370")
+    def test_39370(self):
+        """
```
what if this test runs first and there is no deployment, either you ca combine this test with above one.

ashukakkar replied:
We want this to be different TC.

pragamj commented on libs/prov/prov_k8s_cortx_deploy.py:
```diff
@@ -1728,3 +1723,40 @@ def generate_and_compare_both_version(input_installing_version, installed_versio
+            conf = yaml.safe_load(soln)
+            parent_key = conf['solution']
+            storage_key = parent_key["storage"]
+            cvg_key = storage_key["cvg1"]["devices"]["data"]
```
use variable instead of hardcoding the cvg

ashukakkar replied:
Done.

ashukakkar commented on libs/prov/prov_k8s_cortx_deploy.py:
```diff
@@ -1728,3 +1723,40 @@ def generate_and_compare_both_version(input_installing_version, installed_versio
+        else:
+            # Extend else condition when a different granular deployment type is available.
+            pass
+        storage_key.pop("cvg2")
```
Done.

ashukakkar commented:
Please find below test run html report and console logs after addressing review comments.
[Data only deploy.zip](https://github.com/Seagate/cortx-test/files/8571726/Data.only.deploy.zip)

pragamj commented on libs/prov/prov_k8s_cortx_deploy.py:
```diff
@@ -1728,3 +1723,44 @@ def generate_and_compare_both_version(input_installing_version, installed_versio
+        for cvg in prov_deploy_cfg["cvg_config"]:
+            if cvg == "cvg1":
+                cvg_key = storage_key[cvg]["devices"]["data"]
+                cvg_key.pop("d7")
```
what if i need to have 10 disk per cvg in that case will it work?

pragamj replied:
Please use generic forms so that we can avoid hardcoded values

aayushisharma88 replied:
Done.

pragamj commented on comptests/prov/test_prov_data_only_deploy.py:
```diff
@@ -0,0 +1,148 @@
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Step 1: Clone cortx-k8s script repo on setup.")
+        cmd = "cd {}; {}".format(cons.HA_TMP,
```
use Template string instead of .format

pragamj replied:
do it for all the .format used in this file

seagate-sarang-sawant reviewed (commented):
@ashukakkar  You need to improve the info and debug messages.  Is your test logic complete by just checking the pod status ?

sonalk0209 commented on comptests/prov/test_prov_data_only_deploy.py:
```diff
@@ -0,0 +1,149 @@
+        for node_obj in self.host_list:
+            self.deploy_lc_obj.execute_prereq_cortx(node_obj,
+                                                    self.prov_deploy_cfg["git_remote_path"],
+                                                    "/dev/sdb")
```
can we take /dev/sdb/ from yaml just in case it chnages in future?

sonalk0209 commented on comptests/prov/test_prov_data_only_deploy.py:
```diff
@@ -0,0 +1,149 @@
+        Verify cluster status for data only pod deployment.
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Check Cluster health and services.")
```
I guess we are checking only for data pods, so can we specify something like that?

aayushisharma88 replied:
Sure.

sonalk0209 commented on comptests/prov/test_prov_data_only_deploy.py:
```diff
@@ -0,0 +1,149 @@
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Check Cluster health and services.")
+        resp = self.deploy_lc_obj.check_s3_status(self.master_node_obj)
```
in data pod, we won't be checking s3 server but only ioservices so either method call needs different or the method name needs to be different or maybe generic

aayushisharma88 replied:
Changed Method name as generic.

sonalk0209 replied:
The issue here is @aayushisharma88 that with only data pod deployment, do u really get s3 status or just status for io services which are part of data pod?

pragamj replied:
we have check_service_status method to check for services. and its generic  @aayushisharma88  please use this method.

aayushisharma88 replied:
Ok Sure.

sonalk0209 commented on libs/prov/prov_k8s_cortx_deploy.py:
```diff
@@ -1751,6 +1748,61 @@ def generate_and_compare_both_version(input_installing_version, installed_versio
+        :return: True/False and local file
+        """
+        cvg_count = kwargs.get("cvg_count", 2)
+        data_disk_per_cvg = kwargs.get("data_disk_per_cvg", 0)
```
default 0 is allowed for deployment?

aayushisharma88 replied:
In prov_k8_cortx_deploy.py class there is one method update_cvg_sol_file so from there i reffered it and make logic according to my data disk per cvg.

sonalk0209 replied:
@pragamj is 0 value for data disk is allowed?

pragamj replied:
if its 0 it will pick all the available disk in the node.

sonalk0209 commented on comptests/prov/test_prov_data_only_deploy.py:
```diff
@@ -0,0 +1,149 @@
+    @pytest.mark.tags("TEST-39333")
+    def test_39333(self):
+        """
+        Verify S3 status for all pods.
```
are we verifying s3 status here or just ioservice status?

aayushisharma88 replied:
s3 status

sonalk0209 reviewed (approved):
Approving for merge.

sonalk0209 commented on comptests/prov/test_prov_data_only_deploy.py:
```diff
@@ -0,0 +1,149 @@
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Check Cluster status for data pods")
+        resp = self.deploy_lc_obj.check_service_status(self.master_node_obj)
```
To use this, I guess you would need to set deployment_type to data-only somewhere, isn't it @pragamj ??

pragamj replied:
yes

pragamj replied:
its reading from environment variable(DEPLOYMENT_TYPE).
